test/span.js

```diff
@@ -185,31 +184,80 @@ describe('span should', () => {
+            // We ignore the following case because setOperationName can only go from a sampling decision of 0->1
+            // makeTestCase(0, true, true, "poriyal", false, "murungakkai", false),
+            makeTestCase(0, true, false, "poriyal", true, "murungakkai", true),
+            makeTestCase(0, true, false, "poriyal", false, "murungakkai", false),
+
+            // Never call the sampler when the span is debug or sampled
+            makeTestCase(constants.SAMPLED_MASK, false, true, "avial", null, null, true),
+            makeTestCase(constants.SAMPLED_MASK, false, false, "avial", null, null, true),
+            makeTestCase(constants.SAMPLED_MASK, false, true, "avial", true, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK, false, true, "avial", false, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK, false, false, "avial", true, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK, false, false, "avial", false, "puri", true),
+            
+            makeTestCase(constants.DEBUG_MASK, false, true, "avial", null, null, true),
+            makeTestCase(constants.DEBUG_MASK, false, false, "avial", null, null, true),
+            makeTestCase(constants.DEBUG_MASK, false, true, "avial", true, "puri", true),
+            makeTestCase(constants.DEBUG_MASK, false, true, "avial", false, "puri", true),
+            makeTestCase(constants.DEBUG_MASK, false, false, "avial", true, "puri", true),
+            makeTestCase(constants.DEBUG_MASK, false, false, "avial", false, "puri", true),
+            
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, true, "avial", null, null, true),
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, false, "avial", null, null, true),
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, true, "avial", true, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, true, "avial", false, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, false, "avial", true, "puri", true),
+            makeTestCase(constants.SAMPLED_MASK | constants.DEBUG_MASK, false, false, "avial", false, "puri", true),
+        ];
+
+        testCases.forEach(function(testCase) {
+            it(testCase.name, function() {
+                spanContext._flags = testCase.flags;
+
+                let mockSampler = sinon.mock(tracer._sampler);
+                if (testCase.callSampler) {
+                    mockSampler.expects('isSampled').withExactArgs(testCase.original.operationName, {})
+                        .returns(testCase.original.samplingDecision);
+                }
+
+                let child = tracer.startSpan(testCase.original.operationName, {childOf: span.context()});
+
+                if (testCase.callSampler && testCase.new.operationName) {
```

Comment:
shouldn't `callSampler` be a property of each of `original` and `new`? Are there scenarios when it's called the first time but not the second?